### PR TITLE
Using coffeelint support

### PR DESCRIPTION
**Pull request by torn4dom4n**
Replace `grunt-coffee` with `coffeelint` and fix lint.

### CODE REVIEW

File: spec/text-buffer-spec.coffee [hunk 1]
```diff
@@ -2081,15 +2081,15 @@ describe "TextBuffer", ->
...
+        buffer.scanInRange /cuRRent/i, [[0, 0], [12, 0]], ({match, range}) ->
           matches.push(match)
         expect(matches.length).toBe 1
 
     describe "when given a regex with no global flag", ->
       it "calls the iterator with the first match for the given regex in the given range", ->
         matches = []
         ranges = []
-        buffer.scanInRange /cu(rr)ent/, [[4,0], [6,44]], ({match, range}) ->
+        buffer.scanInRange /cu(rr)ent/, [[4, 0], [6 ,44]], ({match, range}) ->
```
Comment (50Wliu) [inline spec/text-buffer-spec.coffee]: whoops

Comment (torn4dom4n) [inline spec/text-buffer-spec.coffee]: Done!

File: src/match-iterator.coffee [hunk 2]
```diff
@@ -13,7 +13,7 @@ class SingleLineSearchCallbackArgument
 
     set: (@computedRange) ->
 
-  Object.defineProperty @::, 'lineText',
+  Object.defineProperty SingleLineSearchCallbackArgument.prototype, 'lineText',
```
Comment (50Wliu) [inline src/match-iterator.coffee]: Pretty sure it should be `this.prototype`

Comment (torn4dom4n) [inline src/match-iterator.coffee]: Thanks! I am so stupid!

**Comment by torn4dom4n** [comment]
@50Wliu I finished my job.

**Comment by 50Wliu** [comment]
Looks good.

**Pull request merged**